defender/downloaded/Datadog_Alerts_Heat_Map.js

```diff
@@ -0,0 +1,267 @@
+/* eslint-disable import/no-unresolved,import/no-extraneous-dependencies */
+const axios = require('axios');
+const { KeyValueStoreClient } = require('defender-kvstore-client');
+/* eslint-enable import/no-unresolved,import/no-extraneous-dependencies */
+
+// this value was retrieved from the Forta Explorer front-end
+// it corresponds to 05-DEC-2021, presumably the launch date of Forta Explorer
+const fortaExplorerEarliestTimestamp = 1638721490212;
+
+const botIdsToNames = {
+  '0x5a00b44b2db933d4c797e6bd3049abdeb89cc9ec1b2eaee7bdbaff911794f714': 'Low Liquidity Attack',
+  '0xb6bdedbae67cc82e60aad02a8ffab3ccbefeaa876ca7e4f291c07c798a95e339': 'Large Borrows Governance',
+  '0x916603512086fcad84c35858d2fc5356c512f72b19c80e52e8f9c04d8122e2ba': 'Community Multi-Sig',
+  '0x0d3cdcc2757cd7837e3b302a9889c854044a80835562dc8060d7c163fbb69d53': 'Large Delegations',
+  '0xe200d890a67d51c3610520dd9fdfa9e2bd6dd341d41e32fa457601e73c4c6685': 'Oracle Price',
+  '0xf836bda7810aa2dd9df5bb7ac748f173b945863e922a15bb7c57da7b0e6dab05': 'Underlying Asset',
+  '0x125c36816fbad9974a452947bf6a98d975988ddf4342c159a986383b64765e22': 'Market Activity',
+  '0xa0424dfee87cc34b9ff6a1dfa2cb22dbf1b20a238698ae0eeffbf07f869e5b39': 'Governance Activity',
+};
+
+const botIds = Object.keys(botIdsToNames);
+
+const fortaExplorerApiEndpoint = 'https://explorer-api.forta.network/graphql';
+const datadogApiEndpoint = 'https://api.datadoghq.com/api/v2/series';
+
+function camelize(str, delimiter) {
+  let output = '';
+  str.split(delimiter).forEach((element, index) => {
+    const add = element.toLowerCase();
+    if (index === 0) {
+      output += add;
+    } else {
+      output += add[0].toUpperCase() + add.slice(1);
+    }
+  });
+  return output;
+}
+
+// extract relevant information from the Forta Explorer response and create an Object that can
+// be submitted to the Datadog Events API endpoint
+function parseAlertsResponse(response) {
+  const { data: { data: { getList: { alerts } } } } = response;
+  const newAlerts = alerts.map((alert) => {
+    const {
+      protocol,
+      severity,
+      source: {
+        agent: {
+          id: botId,
+        },
+        block: {
+          timestamp: dateTimestamp,
+        },
+      },
+    } = alert;
+
+    const timestamp = new Date(dateTimestamp).valueOf();
+
+    // force the x-axis value (timestamp) passed with the data to be the timestamp at the start of the week
+    const weeksElapsed = Math.floor((timestamp - fortaExplorerEarliestTimestamp) / (1000 * 60 * 60 * 24 * 7));
```

Comment:
```suggestion
    // force the x-axis value (timestamp) passed with the data to be the timestamp at the start
    // of the week
    const msPerWeek = (1000 * 60 * 60 * 24 * 7);
    const weeksElapsed = Math.floor((timestamp - fortaExplorerEarliestTimestamp) / msPerWeek);
```
These lines are longer than 100 characters.

Reply:
changed